apiserver/facades/client/client/status.go

```diff
@@ -1290,9 +1290,13 @@ func (context *statusContext) processApplication(application *state.Application)
 		Life:             processLife(application),
 	}
 
-	if latestCharm, ok := context.allAppsUnitsCharmBindings.latestCharms[*applicationCharm.URL().WithRevision(-1)]; ok && latestCharm != nil {
-		if latestCharm.Revision() > applicationCharm.URL().Revision {
-			processedStatus.CanUpgradeTo = latestCharm.String()
+	curl, err := charm.ParseURL(applicationCharm.URL())
+	if err != nil {
+		return params.ApplicationStatus{Err: apiservererrors.ServerError(err)}
+	}
+	if latestCharm, ok := context.allAppsUnitsCharmBindings.latestCharms[*curl.WithRevision(-1)]; ok && latestCharm != nil {
+		if latestCharm.Revision() > curl.Revision {
+			processedStatus.CanUpgradeTo = latestCharm.URL()
```

Comment:
This will be interesting for planning the removal of charm url - there are implication for the charmrevisionupdaterworker